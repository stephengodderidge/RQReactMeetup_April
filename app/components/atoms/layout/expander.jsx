import styled from 'styled-components';

export const Expander = styled.div`
  margin: auto;
`;
