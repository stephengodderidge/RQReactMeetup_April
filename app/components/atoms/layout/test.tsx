import React from 'react';
import { render } from 'react-testing-library';
import {
  Row,
  Column,
  InjectElement,
  Spacer,
  JustifyContent,
  LayoutBgColor,
} from '.';
import { Expander } from './expander';

describe('Layout', () => {
  // #region Row
  describe('Row', () => {
    test('Renders', () => {
      const { container } = render(
        <Row>
          <div>Element 1</div>
          <div>Element 2</div>
        </Row>,
      );
      expect(container).toMatchSnapshot();
    });
    test('Renders Evenly Spaced Children', () => {
      const { container } = render(
        <Row childSpacing={16}>
          <div>Element 1</div>
          <div>Element 2</div>
        </Row>,
      );
      expect(container).toMatchSnapshot();
    });
    test('Renders - One of Each - Content Justification', () => {
      const oneOfEach = Object.keys(JustifyContent).map(
        (bgColor: keyof typeof JustifyContent) => (
          <Row justifyContent={JustifyContent[bgColor]} key={bgColor}>
            <div>Element 1</div>
            <div>Element 2</div>
          </Row>
        ),
      );
      const { container } = render(<>{oneOfEach}</>);
      expect(container).toMatchSnapshot();
    });
    test('Renders - One of Each - LayoutBgColor', () => {
      const oneOfEach = Object.keys(LayoutBgColor).map(
        (bgColor: keyof typeof LayoutBgColor) => (
          <Row bgColor={LayoutBgColor[bgColor]} key={bgColor}>
            <div>Element 1</div>
            <div>Element 2</div>
          </Row>
        ),
      );
      const { container } = render(<>{oneOfEach}</>);
      expect(container).toMatchSnapshot();
    });
  });
  // #endregion Row
  // #region Column
  describe('Column', () => {
    test('Renders', () => {
      const { container } = render(
        <Column>
          <div>Element 1</div>
          <div>Element 2</div>
        </Column>,
      );
      expect(container).toMatchSnapshot();
    });
    test('Renders Evenly Spaced Children', () => {
      const { container } = render(
        <Column childSpacing={16}>
          <div>Element 1</div>
          <div>Element 2</div>
        </Column>,
      );
      expect(container).toMatchSnapshot();
    });
    test('Renders - One of Each - Content Justification', () => {
      const oneOfEach = Object.keys(JustifyContent).map(
        (bgColor: keyof typeof JustifyContent) => (
          <Row justifyContent={JustifyContent[bgColor]} key={bgColor}>
            <div>Element 1</div>
            <div>Element 2</div>
          </Row>
        ),
      );
      const { container } = render(<>{oneOfEach}</>);
      expect(container).toMatchSnapshot();
    });
    test('Renders - One of Each - LayoutBgColor', () => {
      const oneOfEach = Object.keys(LayoutBgColor).map(
        (bgColor: keyof typeof LayoutBgColor) => (
          <Row bgColor={LayoutBgColor[bgColor]} key={bgColor}>
            <div>Element 1</div>
            <div>Element 2</div>
          </Row>
        ),
      );
      const { container } = render(<>{oneOfEach}</>);
      expect(container).toMatchSnapshot();
    });
  });
  // #endregion Column
  // #region Inject Element
  describe('Inject Element', () => {
    test('Renders', () => {
      const { container } = render(
        <InjectElement element={<div />}>
          <div>Element 1</div>
          <div>Element 2</div>
          <div>Element 3</div>
        </InjectElement>,
      );
      expect(container).toMatchSnapshot();
    });
    test('Injects Elements', () => {
      const testId = 'injected-div';
      const { getAllByTestId } = render(
        <InjectElement element={<div data-testid={testId} />}>
          <div>Element 1</div>
          <div>Element 2</div>
          <div>Element 3</div>
        </InjectElement>,
      );
      expect(getAllByTestId(testId).length).toEqual(2);
    });
  });
  // #endregion Inject Element
  // #region Spacer
  describe('Spacer', () => {
    test('Renders w/ Width', () => {
      const { container } = render(<Spacer width={16} />);
      expect(container).toMatchSnapshot();
    });
    test('Renders w/ Height', () => {
      const { container } = render(<Spacer height={16} />);
      expect(container).toMatchSnapshot();
    });
    test('Renders w/ Width + Height', () => {
      const { container } = render(<Spacer width={16} height={16} />);
      expect(container).toMatchSnapshot();
    });
  });
  // #endregion Spacer
  // #region Expander
  describe('Expander', () => {
    test('Renders w/ Width', () => {
      const { container } = render(<Expander />);
      expect(container).toMatchSnapshot();
    });
  });
  // #endregion Expander
});
