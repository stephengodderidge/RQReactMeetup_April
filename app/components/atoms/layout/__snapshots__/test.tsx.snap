// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Column Renders - One of Each - Content Justification 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
  padding: false;
}

.c2 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  padding: false;
}

.c3 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: transparent;
  padding: false;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c2 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c3 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Column Renders - One of Each - LayoutBgColor 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
  padding: false;
}

.c2 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #f5f6fa;
  padding: false;
}

.c3 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #3867d6;
  padding: false;
}

.c4 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #D3D9E1;
  padding: false;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c2 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c3 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c4 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Column Renders 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
  padding: false;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Column Renders Evenly Spaced Children 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 16px;
}

<div>
  <div
    class="c0"
  >
    <div>
      Element 1
    </div>
    <div
      class="c1"
      height="16"
    />
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Expander Renders w/ Width 1`] = `
.c0 {
  margin: auto;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Layout Inject Element Renders 1`] = `
<div>
  <div>
    Element 1
  </div>
  <div />
  <div>
    Element 2
  </div>
  <div />
  <div>
    Element 3
  </div>
</div>
`;

exports[`Layout Row Renders - One of Each - Content Justification 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
  padding: false;
}

.c2 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  padding: false;
}

.c3 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: transparent;
  padding: false;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c2 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c3 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Row Renders - One of Each - LayoutBgColor 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
  padding: false;
}

.c2 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #f5f6fa;
  padding: false;
}

.c3 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #3867d6;
  padding: false;
}

.c4 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #D3D9E1;
  padding: false;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c2 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c3 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c4 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Row Renders 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
  padding: false;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      Element 1
    </div>
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Row Renders Evenly Spaced Children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 16px;
}

<div>
  <div
    class="c0"
  >
    <div>
      Element 1
    </div>
    <div
      class="c1"
      width="16"
    />
    <div>
      Element 2
    </div>
  </div>
</div>
`;

exports[`Layout Spacer Renders w/ Height 1`] = `
.c0 {
  height: 16px;
}

<div>
  <div
    class="c0"
    height="16"
  />
</div>
`;

exports[`Layout Spacer Renders w/ Width + Height 1`] = `
.c0 {
  width: 16px;
  height: 16px;
}

<div>
  <div
    class="c0"
    height="16"
    width="16"
  />
</div>
`;

exports[`Layout Spacer Renders w/ Width 1`] = `
.c0 {
  width: 16px;
}

<div>
  <div
    class="c0"
    width="16"
  />
</div>
`;
